### ISSUE TITLE
[BUG]: Intervals don't update after changing Desired Retention

### ISSUE THREAD

**Issue by brishtibheja** [open-spaced-repetition/fsrs4anki#657]
My `desired retention` retention was previously set at 77 percent which I then update to 90 percent. I don't use `reschedule cards at change`. Now the intervals shown on top of `hard`, `good` and `easy` buttons should change. Note that I'm speaking of the intervals that each grade will assign to the cards and not the already assigned intervals. They are shown to you by default in the review screen.

Here's where I think the anomaly lies — the same intervals shown for one of my relearning cards doesn't change at all. The said card has gone through the default steps `1m 10m` and `FSRS` determines now what happens at the last step. It feels as if the change in `desired retention` isn't being taken into account. 

Note: The card has gone through relearning steps several days and I just never "graduated" it.

**Environment**
 - Anki version: 24.04.01
 - OS: Android
 

**Additional context**

**Comment by L-M-Sherlock** [comment]
Could you share the card info?

**Comment by brishtibheja** [comment]
I can do more —
![Screenshot_2024-06-17-15-01-19-64_a9eef3a2a561b80d5c76daebd0f9a14c](https://github.com/open-spaced-repetition/fsrs4anki/assets/136738526/489e8109-957f-4bbc-aad0-ee9e8097bcc2)

___

[漢字 Writing__Level 10-4__4. Level 6-20240617133543.zip](https://github.com/user-attachments/files/15864752/Writing__Level.10-4__4.Level.6-20240617133543.zip)

(rename `.zip` —> `.apkg`)

Expertium pinged you in Discord but I thought this would be better place if it is in fact a bug (you are also really active here).

**Comment by L-M-Sherlock** [comment]
The card id, please. I don't know which one is affected.

**Comment by brishtibheja** [comment]
I think that should not be needed as there is only one card in relearning and none in learning. But just in case I missed something, here you go — `cid:1564439362436`

**Comment by L-M-Sherlock** [comment]
I get it. The interval has been determined by the last review.

**Comment by L-M-Sherlock** [comment]
If the card's type is relearn, and the next state is review, the interval is card.interval, which is determined by the last review.

https://github.com/ankitects/anki/blob/33a923797afc9655c3b4f79847e1705a1f998d03/rslib/src/scheduler/answering/current.rs#L125-L132

**Comment by L-M-Sherlock** [comment]
This bug should be fixed in the Anki client's part. I recommend opening a new issue in https://github.com/ankitects/anki

**Comment by brishtibheja** [comment]
It is also hard to notice. Not many people change their settings in the midst studying.

Also I opened another issue but dae has been out for a week now so it will probably be a while before he looks into this.

**Comment by L-M-Sherlock** [comment]
A temporary solution is to use the helper add-on to reschedule the card after you rate it.

<img width="353" alt="image" src="https://github.com/open-spaced-repetition/fsrs4anki/assets/32575846/5ce7b5a1-e74d-4471-b076-2ffab5d0ed86">

**Comment by user1823** [comment]
Many times, I move cards to different decks with different FSRS parameters after rating them `Again`. So, does this issue mean that the interval after pressing `Good` on that relearning card is based on the stability and desired retention of the card when it was in the earlier preset?

**Comment by L-M-Sherlock** [comment]
> So, does this issue mean that the interval after pressing `Good` on that relearning card is based on the stability and desired retention of the card when it was in the earlier preset?

Yes. That's the issue. I'm fixing it in: https://github.com/ankitects/anki/pull/3236
